static/js/dist/app/settings.min.js

```diff
@@ -1 +1,233 @@
+            })
+            .always(function (data){
+                document.body.scrollTop = 0;
+                document.documentElement.scrollTop = 0;
+            })
+        
+        return false
+    })
+
+    $("#validateimap").click(function() {
+
+        // Query validate imap server endpoint
+        var server = {}
+        server.host = $("#imaphost").val()
+        server.port = $("#imapport").val()
+        server.username = $("#imapusername").val()
+        server.password = $("#imappassword").val()
+        server.tls = $('#use_tls').prop('checked')
+
+        //To avoid unmarshalling error in controllers/api/imap.go. It would fail gracefully, but with a generic error. 
+        if (server.host == ""){
+            errorFlash("No IMAP Host specified")
+            document.body.scrollTop = 0;
+            document.documentElement.scrollTop = 0;
+            return false
+        }
+        if (server.port == ""){
+            errorFlash("No IMAP Port specified")
+            document.body.scrollTop = 0;
+            document.documentElement.scrollTop = 0;
+            return false
+        }
+        if (isNaN(server.port) || server.port <1 || server.port > 65535  ){
+            errorFlash("Invalid IMAP Port")
+            document.body.scrollTop = 0;
+            document.documentElement.scrollTop = 0;
+            return false
+        }
+
+        var oldHTML = $("#validateimap").html();
+        // Disable inputs and change button text
+        $("#imaphost").attr("disabled", true);
+        $("#imapport").attr("disabled", true);
+        $("#imapusername").attr("disabled", true);
+        $("#imappassword").attr("disabled", true);
+        $("#use_imap").attr("disabled", true);
+        $("#use_tls").attr("disabled", true);
+        $("#folder").attr("disabled", true);
+        $("#restrictdomain").attr("disabled", true);
+        $('#deletecampaign').attr("disabled", true);
+        $('#lastlogin').attr("disabled", true);
+        $('#imapfreq').attr("disabled", true);
+        $("#validateimap").attr("disabled", true);  
+        $("#validateimap").html("<i class='fa fa-circle-o-notch fa-spin'></i> Testing...");
+        
+        api.IMAP.validate(server).done(function(data) {
+            if (data.success == true) {
+                Swal.fire({
+                    title: "Success",
+                    html: "Logged into <b>" + $("#imaphost").val() + "</b>",
+                    type: "success",
+                })
+            } else {
+                Swal.fire({
+                    title: "Failed!",
+                    html: "Unable to login to <b>" + $("#imaphost").val() + "</b>.",
+                    type: "error",
+                    showCancelButton: true,
+                    cancelButtonText: "Close",
+                    confirmButtonText: "More Info",
+                    confirmButtonColor: "#428bca",
+                    allowOutsideClick: false,
+                }).then((result) => {
```

Comment:
Heads up that we don't currently support ES6 constructs in settings.js. It's a WIP to migrate everything over to the new webpack + Babel workflow. The hard part is making sure that we don't break anything in the process due to things like different `this` scoping.

Reply:
Cool updated to `then(function(result) {` syntax.

Also just noticed hitting 'Close' on that dialog wasn't closing it as I wasn't checking the result value, fixed that too.